engine/src/cubos/engine/tools/scene_editor/plugin.cpp

```diff
@@ -0,0 +1,263 @@
+#include <imgui.h>
+#include <misc/cpp/imgui_stdlib.h>
+
+#include <cubos/core/ecs/commands.hpp>
+
+#include <cubos/engine/assets/plugin.hpp>
+#include <cubos/engine/imgui/plugin.hpp>
+#include <cubos/engine/scene/plugin.hpp>
+#include <cubos/engine/tools/asset_explorer/plugin.hpp>
+#include <cubos/engine/tools/entity_selector/plugin.hpp>
+#include <cubos/engine/tools/scene_editor/plugin.hpp>
+
+using cubos::core::ecs::Commands;
+using cubos::core::ecs::Entity;
+using cubos::core::ecs::Read;
+using cubos::core::ecs::Write;
+
+using namespace cubos::engine;
+
+using tools::AssetSelectedEvent;
+
+/// @brief Resource used to store Scene(s) information
+struct SceneInfo
+{
+    bool isExpanded{false};
+    bool isSelected{false};
+    bool shouldBeRemoved{false};
+    std::string name;
+    std::vector<std::pair<std::string, Entity>> entities;
+    std::vector<std::pair<std::string, SceneInfo>> scenes;
+};
+
+static void closeScene(Commands& commands, SceneInfo& scene)
+{
+    for (const auto& [name, entity] : scene.entities)
+    {
+        commands.destroy(entity);
+    }
+    scene.entities.clear();
+
+    for (auto& [name, subscene] : scene.scenes)
+    {
+        closeScene(commands, subscene);
+    }
+    scene.scenes.clear();
+}
+
+static void placeEntity(std::string name, Entity handle, SceneInfo& scene)
+{
+    auto split = name.find(".");
+    if (split == std::string::npos)
+    {
+        scene.entities.emplace_back(name, handle);
+    }
+    else
+    {
+        auto subsceneName = name.substr(0, split - 1);
+        for (auto& [sname, subscene] : scene.scenes)
+        {
+            if (sname.compare(subsceneName) == 0)
+            {
+                placeEntity(name.substr(split + 1), handle, subscene);
+                return;
+            }
+        }
+        SceneInfo subScene;
+        subScene.name = subsceneName;
+        auto& [_, subSceneHandle] = scene.scenes.emplace_back(subsceneName, subScene);
+        placeEntity(name.substr(split + 1), handle, subSceneHandle);
+    }
+}
+
+static void openScene(const Asset<Scene>& sceneToSpawn, Commands& commands, const Assets& assets, SceneInfo& scene)
+{
+    closeScene(commands, scene);
+    auto sceneRead = assets.read(sceneToSpawn);
+    auto builder = commands.spawn(sceneRead->blueprint);
+    for (const auto& [entity, name] : sceneRead->blueprint.getMap())
+    {
+        placeEntity(name, builder.entity(name), scene);
+    }
+}
+
+static void checkAssetEventSystem(cubos::core::ecs::EventReader<AssetSelectedEvent> reader, Commands commands,
+                                  Read<Assets> assets, Write<SceneInfo> scene)
+{
+    for (const auto& event : reader)
+    {
+        assets->load(event.asset);
+        if (assets->type(event.asset) == typeid(Scene))
+        {
+            CUBOS_TRACE("CAN SPAWN");
```

Comment:
```suggestion
            CUBOS_INFO("Opening scene {}", Debug(event.asset));
```

Dont forget to add  `#include <cubos/core/data/debug_serializer.hpp>` and `using cubos::core::data::Debug`.